lib/src/cli.rs

```diff
@@ -378,7 +379,10 @@ async fn upgrade(opts: UpgradeOpts) -> Result<()> {
             println!("No update available.")
         } else {
             let osname = booted_deployment.osname();
-            crate::deploy::stage(sysroot, &osname, &fetched, &spec).await?;
+            let mut opts = ostree::SysrootDeployTreeOpts::default();
+            let kargs: Vec<&str> = kargs.iter_mut().map(|s| { s.pop(); s.as_str() }).collect();
+            opts.override_kernel_argv = Some(kargs.as_slice());
```

Comment:
The tricky part is...this will *entirely override* the kernel arguments, which means that any machine-specific kargs will get lost.  In the general case we need to support:

- kernel args owned by the image (as this PR is doing)
- kernel args injected as external state; this is what we're doing in coreos with the `console=` kargs for example, but we'll also support doing that via other installation tooling and image generation tooling

Tangential but what's up with the `s.pop()` there?  That looks odd...

Reply:
> Tangential but what's up with the s.pop() there? That looks odd...

Oh I added it in to get rid of the newline character at the end of each karg.

Reply:
Ah, I think we want to [trim()](https://doc.rust-lang.org/std/primitive.str.html#method.trim) then.